.github/workflows/third_party.yml

```
@@ -0,0 +1,49 @@
+name: Third-party bots
+
+on:
+    push:
+        branches: [main]
+        paths:
+            - src/
+            - .github/workflows/third_party.yml
+    pull_request:
+        branches: [main]
+        paths:
+            - src/
+            - .github/workflows/third_party.yml
+
+jobs:
+    deno:
+        runs-on: ubuntu-latest
+
+        strategy:
+            matrix:
+                url:
+                    # please keep sorted
+                    - https://raw.githubusercontent.com/dcdunkan/ryportbot/main/main.ts
+                    - https://raw.githubusercontent.com/dcdunkan/y-by-example/main/contents/commands.ts
+                    - https://raw.githubusercontent.com/dcdunkan/y-by-example/main/contents/hello-world.ts
+                    - https://raw.githubusercontent.com/dcdunkan/y-by-example/main/contents/inline-keyboards.ts
+                    - https://raw.githubusercontent.com/dcdunkan/y-by-example/main/contents/message-formatting.ts
+                    - https://raw.githubusercontent.com/grammyjs/docs-bot/main/bot.ts
+
+        steps:
+            - uses: denoland/setup-deno@v1
+              with:
+                deno-version: v1.x
+
+            - name: Create import_map.json
+              run: |
+                set -o pipefail;
+                curl --silent https://cdn.deno.land/grammy/meta/versions.json |
+                jq '
+                    (.latest / ".")[0] as $major
+                    | .versions
+                    | map(select(startswith("\($major).")) | {
```

```suggestion
                    | map(select(startswith($major + ".")) | {
```
is probably more readable

---

and less Swift-ish. Nice.